indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -1293,7 +1293,8 @@ private void discoverTasks() throws ExecutionException, InterruptedException, Ti
     int taskCount = 0;
     List<String> futureTaskIds = new ArrayList<>();
     List<ListenableFuture<Boolean>> futures = new ArrayList<>();
-    List<Task> tasks = taskStorage.getActiveTasks();
+    List<Task> tasks = taskStorage.getActiveTasksByDatasource(dataSource);
```

Comment:
I think you can also change line 1301, since it no longer needs to check that the datasource is equal

Reply:
Yes, I will change it.